src/librustc_mir/borrow_check/nll/region_infer.rs

```diff
@@ -33,15 +35,21 @@ pub struct RegionInferenceContext<'tcx> {
     /// regions, these start out empty and steadily grow, though for
     /// each free region R they start out containing the entire CFG
     /// and `end(R)`.
-    liveness_constraints: IndexVec<RegionVid, Region>,
+    liveness_constraints: BitMatrix,
 
     /// The final inferred values of the inference variables; `None`
     /// until `solve` is invoked.
-    inferred_values: Option<IndexVec<RegionVid, Region>>,
+    inferred_values: Option<BitMatrix>,
 
     /// The constraints we have accumulated and used during solving.
     constraints: Vec<Constraint>,
 
+    /// A map from each MIR Location to its column index in
+    /// `liveness_constraints`/`inferred_values`
+    point_indices: BTreeMap<Location, usize>,
```

Comment:
It'd be nice to expand this comment to explain that the first N indices are the free regions -- or at least comment on that *somewhere*.